chainer/optimizer.py

```diff
@@ -518,3 +530,52 @@ def __call__(self, opt):
                 grad = param.grad
                 with cuda.get_device(grad):
                     grad *= rate
+
+
+class GradientNoise(object):
+    """Optimizer hook function for adding gradient noise.
+
+    This hook function simply adds noise generated by the `noise_func`
+    to the gradient. By default it adds time-dependent annealed Gaussian
+    noise to the gradient at every training step:
+
+    . math::
+
+        g_t \\leftarrow g_t + N(0, \\sigma_t^2)
+
+    where
+
+    . math::
+
+        \\sigma_t^2 = \\frac{\\eta}{(1+t)^\\gamma}
+
+    with :math:`\\eta` selected from {0.01, 0.3, 1.0} and
+    :math:`\\gamma = 0.55`.
+
+    Args:
+        eta (float): parameter for the noise function, which for the default
```

Comment:
Could you explain the role of the attribute in more detail? For example, "parameter that defines the scale of the noise".